src/dask_awkward/lib/core.py

```diff
@@ -872,6 +872,69 @@ def _getitem_outer_int(self, where: int | tuple[Any, ...]) -> Any:
         else:
             return new_scalar_object(hlg, name, meta=new_meta)
 
+    def _getitem_slice_on_zero(self, where: tuple[slice, ...]):
+        # normalise
+        sl: slice = where[0]
+        rest = tuple(where[1:])
+        step = sl.step or 1
+        start = sl.start or 0
+
+        if not self.known_divisions:
+            self.eager_compute_divisions()
+        stop = sl.stop or self.divisions[-1]
+        start = start if start >= 0 else self.divisions[-1] + start
+        stop = stop if stop >= 0 else self.divisions[-1] + stop
+        if step < 0:
+            raise DaskAwkwardNotImplemented("negative step slice on zeroth dimension")
+
+        # setup
+        token = tokenize(self, where)
+        name = f"getitem-{token}"
+        remainder = 0
+        outpart = 0
+        divisions = [0]
+        dask = {}
+        # make low-level graph
+        for i in range(self.npartitions):
+            if start > self.divisions[i + 1]:
+                # first partition not found
+                continue
+            if stop < self.divisions[i] and dask:
+                # no more partitions with valid rows
+                # does **NOT** exit if there are no partitions yet, to make sure there is always
+                # at least one, needed to get metadata of empty output right
+                break
+            slice_start = max(start - self.divisions[i] + remainder, 0)
+            slice_end = min(stop - self.divisions[i], self.divisions[i + 1])
+            if (
+                slice_end == slice_start
+                and (self.divisions[i + 1] - self.divisions[i])
+                and dask
+            ):
+                # in case of zero-row last partition (if not only partition)
+                break
+            dask[(name, outpart)] = (
+                _zero_getitem,
+                (self.name, i),
+                slice(slice_start, slice_end, step),
+                rest,
+            )
+            outpart += 1
+            remainder += (self.divisions[i + 1] - self.divisions[i]) % step
+            divisions.append(
+                (self.divisions[i + 1] - self.divisions[i]) // step + divisions[-1]
+            )
+            remainder = remainder % step
+        hlg = HighLevelGraph.from_collections(name, dask, dependencies=[self])
+        hlg.layers[name].mock = mock_materialized_layer(hlg.layers[name], self.name)
```

Comment:
this attach-a-free-function-as-instance-method is clever! I was thinking we would need to subclass `MaterialzedLayer` and add a `mock` method

Reply:
I was feeling burned from making layers, but yes that would work of course. It's not actually a method, just an attribute here (making a true dynamic method is possible too).

Reply:
done